docs/developer/architecture/centralized-cli.rst

```diff
@@ -149,3 +149,203 @@ Cluster Service Administration
 +----------+---------------+-------------+----------------------------+
 | edit     | dex-config    | name        | open an editor             |
 +----------+---------------+-------------+----------------------------+
+
+Design Choices
+--------------
+
+Two distinct CLI:
+
+- a ``metalk8s`` kubectl plugin with subcommands to interact with Kubernetes
+  API, and Salt API if needed.
+- a ``metalk8sctl`` CLI with subcommands for action that need to interact with
+  the local machine, but may also interact with Kubernetes API and Salt API if
+  needed.
+
+
+``metalk8s`` kubectl plugin
```

Comment:
Is it really part of this design ? For me here we don't really care what kind of kubeconfig user may use no ?
But yes agree we may want to document how to get a kubeconfig (and I think for user it will be a kubeconfig with a token not a certificate, no ?)

Reply:
Well, if it's not part of this one, then it needs to be part of another - otherwise the `kubectl` plugin can't be an acceptable solution for our users, who mainly don't care about K8s itself and its technologies.

Reply:
perhaps worth also to explain how authentication is working with the tool metalk8sctl

Reply:
Hmm we do not have reall authentication in this tool as it's only local action but yes I should add it I agree

Reply:
#2487

Reply:
> One thing we forgot to mention: kubectl authentication is based on kubeconfig files, with certificates, whereas our users will rely on an OIDC provider to authenticate, most of the time.

That's not true, a `KubeConfig` object can have certs embedded, or a token, or HTTP basic auth.

> It would be nice to include in this design an approach for generating the required certs (maybe even the kubeconfig, I think @NicolasT mentioned tools to do that) based on an OIDC-provided identity.

Those tools don't generate certs or anything like that. They generate a `KubeConfig` including the JWT token.